### DISCUSSION TITLE
Random ConnectionError('Error UNKNOWN while writing to socket. Connection lost.')

### DISCUSSION THREAD

**Discussion by andrijan** [django/channels#2072] [Q&A]
Hey there!

I've been using `channels` in production for a while and everything works smoothly. Except when looking at the logs (or Sentry) I see some random `ConnectionError` in my consumers.  

```python
import json

from channels.generic.websocket import AsyncJsonWebsocketConsumer


class Consumer(AsyncJsonWebsocketConsumer):
    async def connect(self):
        self.id = self.scope["url_route"]["kwargs"]["id"]
        self.group_name = f"group_{self.id}"
        if not self.channel_layer:
            return

        await self.channel_layer.group_add(self.group_name, self.channel_name)  # Here the error is triggered
        await self.accept()

    async def disconnect(self, close_code):
        if not self.channel_layer:
            return
        await self.channel_layer.group_discard(self.group_name, self.channel_name)

    async def update(self, event):
        await self.send(text_data=json.dumps(event))
```

|Screenshot from sentry where the error is triggered in the code above|
|-------------|
|![image](https://github.com/django/channels/assets/250780/157cf6eb-0c59-4589-97de-bf9dabc42739)|

|Screenshot from sentry below the other error|
|-------------|
|![image](https://github.com/django/channels/assets/250780/8386b8c4-65a9-4f6c-a565-a0f9899dd090)|

So my question is basically, are these `ConnectionError`s normal, due to network connectivity issues from the users?

And if so, should I just ignore them (With a `try`/`except` block)?
If not, is there anything I should be doing differently in the consumer or maybe some configuration issue?

I'm using `daphne` and it is running on **Heroku**

**Comment by carltongibson** [comment]
Hi @andrijan. 

Yes... these things can happen. Often there's a timeout somewhere, or something similar, that causes the connection to be closed (or, yes too, there can just be an error.) 

Folks have had success sending a heartbeat every so often, to make sure a connection isn't shut for being idle. 

All this depends on your provider… — I don't know if Heroku have any guidance on long-lived connections. 

It's not really something I can explicitly help you with though, I'm afraid. 

HTH
